test/1d_quadratures.cxx

```diff
@@ -88,218 +52,158 @@ constexpr T ref_gaussian_int( T a, T b ) {
 }
 
 
-constexpr size_t factorial(size_t n) {
-  return (n == 1 || n == 0) ? 1 : factorial(n - 1) * n;
-}
-
-
-template <typename T>
-constexpr std::complex<T> spherical_harmonics( int64_t l, int64_t m, T theta, T phi ) {
-
-
-  auto prefactor = std::sqrt( (2.*l + 1) / (4.*M_PI) * T(factorial(l-m))/T(factorial(l+m)) );
-  if( m < 0 )
-    prefactor *= std::pow(-1,std::abs(m)) * T(factorial(l-m))/T(factorial(l+m));
-
-
-  return prefactor * assoc_legendre( l, std::abs(m), std::cos(theta) ) *
-         std::complex<T>( std::cos(m*phi), std::sin(m*phi) );
-
-}
-
 template <typename T>
-constexpr std::complex<T> spherical_harmonics( int64_t l, int64_t m, T x, T y, T z ) {
-
-  const auto r = std::sqrt( x*x + y*y + z*z );
-
-  return spherical_harmonics( l, m, std::acos( z / r ), std::atan2( y, x ) );
-
+auto chebyshev_T(int n, T x) {
+  return std::cos( n * std::acos(x) );
 }
 
 
-template <typename T, typename... Args>
-constexpr T real_spherical_harmonics( int64_t l, int64_t m, Args&&... args ) {
-
-  const T phase = std::sqrt(2.) * ((m % 2) ? 1. : -1.);
 
-  if( m == 0 )
-    return std::real(spherical_harmonics( l, m, std::forward<Args>(args)... ));
-  else if( m < 0 )
-    return phase * std::imag(spherical_harmonics( l, std::abs(m), std::forward<Args>(args)... ));
-  else
-    return phase * std::real(spherical_harmonics( l, m, std::forward<Args>(args)... ));
 
-}
 
 
 TEST_CASE( "Gauss-Legendre Quadratures", "[1d-quad]" ) {
 
-  for(unsigned order=10;order<14;order++) {
-    std::ostringstream oss;
-    oss << "order " << order;
-    SECTION( oss.str()) {
-
-      IntegratorXX::GaussLegendre<double,double> quad( order );
-
-      const auto& pts = quad.points();
-      const auto& wgt = quad.weights();
-
-      auto f = [=]( double x ){ return gaussian(x); };
+  // Reference integral for polynomial evaluated over [-1,1]
+  auto ref_value = [](const std::vector<double>& c) {
+    const auto p = c.size();
+    std::vector<double> cp(p+1, 0.0); 
+    for(int i = 0; i < p; ++i) {
+      cp[i] = c[i] / (p-i);
+    }
+    return Polynomial::evaluate(cp, 1.0) - Polynomial::evaluate(cp, -1.0);
+  };
 
-      double res = 0.;
-      for( auto i = 0; i < quad.npts(); ++i ) {
-        res += wgt[i] * f(pts[i]);
-      }
+  // Test Quadrature for Correctness
+  using quad_type = IntegratorXX::GaussLegendre<double,double>;
+  test_random_polynomial<quad_type, Polynomial>("Gauss-Legendre", 10, 14, 
+    [](int o){ return 2*o+1; }, // Max order 2N-1
```

Comment:
`2o+1` vs `2N-1`?

Reply:
The `p` parameter generates a polynomial of order `p-1`, and the `p`-loop is not left-inclusive, hence the "off-by-2" here and elsewhere.